cpp/src/arrow/csv/writer.cc

```diff
@@ -195,11 +199,17 @@ class QuotedColumnPopulator : public ColumnPopulator {
     VisitArrayDataInline<StringType>(
         *input.data(),
         [&](arrow::util::string_view s) {
-          int64_t escaped_count = CountEscapes(s);
-          // TODO: Maybe use 64 bit row lengths or safe cast?
-          row_needs_escaping_[row_number] = escaped_count > 0;
-          row_lengths[row_number] += static_cast<int32_t>(s.length()) +
-                                     static_cast<int32_t>(escaped_count + kQuoteCount);
+          if (escaping_) {
+            int64_t escaped_count = CountEscapes(s);
```

Comment:
If escaping is disabled, perhaps we should still count escapes and error if the number is non-zero?